java/team25core/RingDetectionTask.java

```diff
@@ -0,0 +1,279 @@
+            return RingKind.SINGLE_KIND;
+        } else {
+            return RingKind.UNKNOWN_KIND;
+        }
+    }
+    //if recognize anything will add to que
+    protected void processEverything(List<Recognition> objects)
+    {
+        if (objects.size() > 0) {
+            robot.queueEvent(new RingDetectionEvent(this, EventKind.OBJECTS_DETECTED, objects));
+        }
+    }
+    //only adds rings which will make event and add to que
+    protected void processQuadRing(List<Recognition> objects)
+    {
+        List<Recognition> rings = new ArrayList<>();
+        for (Recognition object : objects) {
+            if (isRing(object) == RingKind.QUAD_KIND) {
+                rings.add(object);
+            }
+        }
+
+        if (!rings.isEmpty()) {
+            robot.queueEvent(new RingDetectionEvent(this, EventKind.OBJECTS_DETECTED, rings));
+        }
+    }
+
+    protected void processSingleRing(List<Recognition> objects)
+    {
+        List<Recognition> singlerings = new ArrayList<>();
+        for (Recognition object : objects) {
+            if (isRing(object) == RingKind.SINGLE_KIND) {
+                singlerings.add(object);
+            }
+        }
+
+        if (!singlerings.isEmpty()) {
+            robot.queueEvent(new RingDetectionEvent(this, EventKind.OBJECTS_DETECTED, singlerings));
+        }
+
+    }
+
+    protected void processLargestSkyStone(List<Recognition> objects)
+    {
+        if (objects.isEmpty()) {
+            return;
+        }
+
+        Recognition largest = null;
+        List<Recognition> singleton;
+
+
+        for (Recognition object : objects) {
+            if (isRing(object) == RingKind.SINGLE_KIND) {
+                if (largest == null) {
+                    largest = object;
+                } else if ((largest.getHeight() * largest.getWidth()) < (object.getWidth() * object.getHeight())) {
+                    largest = object;
+                }
+            }
+        }
+
+        if (largest != null) {
+            singleton = new ArrayList<>();
+            singleton.add(largest);
+            robot.queueEvent(new RingDetectionEvent(this, EventKind.OBJECTS_DETECTED, singleton));
+        }
+    }
+
+    //timeslice calls to get information from recognition
+    protected void processDetectedObjects(List<Recognition> objects)
+    {
+        if (objects == null || objects.isEmpty()) {
+            return;
+        }
+
+        switch (detectionKind) {
+            case EVERYTHING:
+                processEverything(objects);
+                break;
+            case SINGLE_RING_DETECTED:
+                processSingleRing(objects);
+                break;
+            case QUAD_RING_DETECTED:
+                processQuadRing(objects);
+                break;
+            case LARGEST_SKY_STONE_DETECTED:
+                processLargestSkyStone(objects);
+                break;
+        }
+    }
+
+    @Override
+    public boolean timeslice()
+
```

Comment:
Your style is generally quite good and consistent.   The only thing I see that could be improved upon is to watch, or look for, extraneous newlines.   Otherwise, quite good.

Reply:
Hi Craig I have made the changes based on your feedback and removed all instances of SkyStone could you please review the code.